explicitely set the 'char *matrix' member of ao_sample_format to NULL
instead of letting it be some random junk.

--- src/plugins/ao/ao.c.orig	Fri Feb 19 09:43:54 2010
+++ src/plugins/ao/ao.c	Sun Feb  6 11:13:39 2011
@@ -166,6 +166,7 @@ xmms_ao_new (xmms_output_t *output)
 		format.rate = 44100;
 		format.channels = 2;
 		format.byte_format = AO_FMT_NATIVE;
+		format.matrix = NULL;
 
 		device = ao_open_live (data->driver_id, &format, data->options);
 		if (!device && errno == AO_EOPENDEVICE) {
@@ -354,6 +355,7 @@ xmms_ao_try_format (gint driver_id, ao_option *options
 	fmt->channels = channels;
 	fmt->rate = samplerate;
 	fmt->byte_format = AO_FMT_NATIVE;
+	fmt->matrix = NULL;
 
 	return TRUE;
 }
